code/drasil-lang/lib/Language/Drasil/Expr/Lang.hs

```diff
@@ -229,17 +224,36 @@ vvvInfer ctx op l r = case (infer ctx l, infer ctx r) of
     (_       , Right re) -> Right re
     (Right le, _       ) -> Right le
 
+
 instance Typed Expr Space where
   check :: TypingContext Space -> Expr -> Space -> Either Space TypeError
   check = typeCheckByInfer
 
   infer :: TypingContext Space -> Expr -> Either Space TypeError
   infer cxt (Lit lit) = infer cxt lit
 
-  infer cxt (AssocA op exs) = allOfType cxt exs sp sp
-      $ "Associative arithmetic operation expects all operands to be of the same expected type (" ++ show sp ++ ")."
-    where
-      sp = assocArithOperToTy op
+  infer cxt (AssocA _ (e:exs)) = 
+    case infer cxt e of
+      Left spaceValue | S.isBasicNumSpace spaceValue -> 
+          -- If the inferred type of e is a valid Space, call allOfType with spaceValue
+          allOfType cxt exs spaceValue spaceValue 
+              "Associative arithmetic operation expects all operands to be of the same expected type."
+      Left _ ->
+          -- Handle the case when sp is a Left value but spaceValue is invalid
+          Right "Invalid spaceValue encountered."
```

Comment:
I think we should change this error message to "Expected all operands in addition/multiplication to be numeric, but found X".